### ISSUE TITLE
Starting a vnc session creates a zombie process

### ISSUE THREAD

**Issue by grulja** [TigerVNC/tigervnc#1148]
**Describe the bug**
When user starts a vnc session, it creates a zombie process of "vncsession".

**To Reproduce**
This can be reproduced easily by starting a vnc session using systemd service or running vncsession directly.

**Expected behavior**
No zombie process.

**Additional context**
This is with Tigervnc 1.11.0.

I have been looking into the code of vncsession and tried to understand how this daemon process should be started. I wonder if there is a bug in closing a wrong process. In the "begin_daemon()" why you close the child right after the second fork, shouldn't it actually close the parent? To me it looks like if you do it this way, you basically create the second child and close it immediately.

Like:
```
diff --git a/unix/vncserver/vncsession.c b/unix/vncserver/vncsession.c
index 2b47f5f5..a4b71423 100644
--- a/unix/vncserver/vncsession.c
+++ b/unix/vncserver/vncsession.c
@@ -99,7 +99,7 @@ begin_daemon(void)
         return -1;
     }
 
-    if (pid == 0)
+    if (pid > 0)
         _exit(0);
 
     /* Send all stdio to /dev/null */
```

**Comment by CendioOssman** [comment]
You're right. It should be `if (pid != 0)` just like for the first `fork()`. Amazed it actually works properly despite that.

Have you tested things with that fix in place?

**Comment by grulja** [comment]
> You're right. It should be `if (pid != 0)` just like for the first `fork()`. Amazed it actually works properly despite that.
> 
> Have you tested things with that fix in place?

Yes. It works just fine as before. Feel free to push the fix directly without me creating a pull request.

**Comment by CendioOssman** [comment]
Thanks. Fixed in fd52f3c538bc6b7ee1bb6d8305c23ae27fab9279.
